### PR TITLE
LRQA-75853, LRQA-75854, LRQA-75855, LRQA-75856, LRQA-75857, LRQA-75858, LRQA-75859, LRQA-75860 | master

### PR DESCRIPTION
**Pull request by Kvale1733**
https://issues.liferay.com/browse/LRQA-75853
https://issues.liferay.com/browse/LRQA-75854
https://issues.liferay.com/browse/LRQA-75855
https://issues.liferay.com/browse/LRQA-75856
https://issues.liferay.com/browse/LRQA-75857
https://issues.liferay.com/browse/LRQA-75858
https://issues.liferay.com/browse/LRQA-75859
https://issues.liferay.com/browse/LRQA-75860

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 1]
```diff
@@ -11,6 +11,8 @@ definition {
 
 	setUp {
 		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:15]: Its fine this time, but this should probably be seperate into a different commit since the intention isnt the same as the commit message itself

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/portalinstance/HeadlessPortalInstanceAPI.macro [hunk 2]
```diff
@@ -1,5 +1,22 @@
 definition {
 
+	@summary = "Get the groupId for a given company"
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/portalinstance/HeadlessPortalInstanceAPI.macro:3]: This is technically wrong, since there could be many groups(sites) within a given company, can you revise the summary and macro name? This is retrieving the default group from a company (the one that is first created when you created a new virtual instance)

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 3]
```diff
@@ -22,75 +24,398 @@ definition {
...
+				userEmailAddress = "test@www.baker.com",
+				virtualHostsURL = "http://www.baker.com:8080");
+
+			ApplicationsMenu.gotoSite(key_site = "www.baker.com");
+
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "Test Test");
+
+			PortalInstances.openVirtualInstancesAdmin();
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:62]: We shouldn't be grabbing the companyId through the UI since we can grab it through the backend API, this will be faster and more stable, see JSONCompany.getCompanyId

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 4]
```diff
@@ -22,75 +24,398 @@ definition {
...
+				fail("logs/companies folder exists");
+			}
+		}
 	}
 
 	@description = "An Omniadmin can get log names from all companies and download logs from all companies though the UI"
-	@ignore = "Test Stub"
 	@priority = "4"
 	test GetAllCompanyLogs {
+		task ("Given two new virtual instances are created") {
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:148]: Same feedback as the other comment related to the virtual instance setup, remove un-necessary steps and utilize the backend when available to retrieve data to make the setup more robust and quick

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 5]
```diff
@@ -22,75 +24,398 @@ definition {
...
+		}
+
+		task ("And the log files contain logs for the specific company") {
+			for (var companies : list "liferay.com:${companyId1},www.able.com:${companyId2},www.baker.com:${companyId3}") {
+				var companyTitle = StringUtil.extractFirst("${companies}", ":");
+				var companyId = StringUtil.extractLast("${companies}", ":");
+
+				var groupId = HeadlessPortalInstanceAPI._getPortalInstanceGroupId(companyId = "${companyId}");
+
+				var groupId = ${groupId} - 2;
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:225]: same as previous, why are we subtacting 2?

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 6]
```diff
@@ -22,75 +24,398 @@ definition {
...
+				domain = "www.able.com",
+				portalInstanceId = "www.able.com",
+				virtualHost = "www.able.com");
+		}
+
+		task ("And given the admin for that virtual instance logs in") {
+			User.logoutPG();
 
-		// TODO LRQA-75855
+			User.firstLoginPG(
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:255]: The test user is an omniadmin, not a company admin. A company admin has permissions only to a specific company. You will need to create a new user and login as a user, there should be other tests that utilize company admin in poshi to use as reference

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 7]
```diff
@@ -22,75 +24,398 @@ definition {
...
+				userEmailAddress = "test@www.able.com",
+				virtualHostsURL = "http://www.able.com:8080");
+		}
+
+		task ("When the company admin tries accessing the company logging UI to download a specific company log from a different company") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "o/company-log/${companyId}/${logName}?download=true");
+		}
+
+		task ("Then a permissions error is shown and no log is downloaded") {
+			AssertTextPresent(
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:312]: This shouldn't be happening since you are logging in with the omni-admin.(They have access to every company). This could either be a bug, or the url we are navigating to here is incorrect if this is passing

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 8]
```diff
@@ -22,75 +24,398 @@ definition {
...
+			JSONUser.setFirstPassword(
+				agreeToTermsAndAnswerReminderQuery = "true",
+				requireReset = "false",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("When the user tries accessing the company logging UI to download a specific company log") {
+			PortalInstances.openVirtualInstancesAdmin();
+
+			var companyId = TestUtils.getElementTextContent(element = "//tr[contains(.,'liferay.com')]/td[1]//a");
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:341]: Use the backend to get the companyId, not the frontend

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 9]
```diff
@@ -22,75 +24,398 @@ definition {
...
+
+			FileUtil.delete("${logsDir}");
+		}
 
+		task ("When the user navigates to the company logging UI access point") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "o/company-log");
+		}
+
+		task ("Then a message is shown that indicates no logs are available") {
+			AssertElementPresent(
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:391]: Since we will likely never re-use this path again (This path just needs to get the message content from the UI), you can hard code the locator into here. Creating a path that will not be re-used is like creating a function that will not be re-used

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase [hunk 10]
```diff
@@ -22,75 +24,398 @@ definition {
...
 
 	@description = "If a user tries to download a logging file or company that doesn’t exist it should fail and throw a suitable error message"
-	@ignore = "Test Stub"
 	@priority = "4"
 	test GetInvalidCompanyLogs {
+		task ("When the user tries to download a log via an http request where the log name is invalid") {
+			PortalInstances.openVirtualInstancesAdmin();
 
-		// TODO LRQA-75860
+			var companyId = TestUtils.getElementTextContent(element = "//tr[contains(.,'liferay.com')]/td[1]//a");
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/logging/CompanyLogging.testcase:403]: use backend

**Comment by kyle-miho** [comment]
@Kvale1733 See my comments, make the necessary fixes and then resend afterwards
